independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -407,6 +410,29 @@ ValidationContext validate(List<BeanDeploymentValidator> validators,
         return Collections.unmodifiableList(beans);
     }
 
+    public Collection<BeanInfo> getBeansByType(Type type) {
+        Map<Type, List<BeanInfo>> byType = beansByType;
+        if (byType == null) {
+            synchronized (this) {
+                byType = beansByType;
```

Comment:
I don't think it makes sense to init the map lazily. I'd prefill the map in the `BeanDeployment.init()` method, after the _unused_ beans are removed.

Reply:
I had the same thought before I even read this comment. It would IMO make much more sense to pre-create this map in build time.